src/piadm.sh

```diff
@@ -176,11 +176,68 @@ vcurl() {
 	fi
 }
 
-# Well-known source of SmartOS Platform Images
+# Default well-known source of SmartOS Platform Images
 DEFAULT_URL_PREFIX=https://us-east.manta.joyent.com/Joyent_Dev/public/SmartOS/
+# Default path for piadm's configuration
+PIADM_CONF=/var/piadm/piadm.conf
 
-# Can be overridden by the user's PIADM_URL_PREFIX.
-URL_PREFIX=${PIADM_URL_PREFIX:-${DEFAULT_URL_PREFIX}}
+#
+# (Re)-Configure the default URL (and potentially other things in the future)
+# using the configuration file (in $PIADM_CONF).
+#
+config_check() {
+	# Can't do standalone_only per se as it errors out, but this only
+	# applies to standalone SmartOS.
+	if [[ "$TRITON_CN" == "yes" || "$TRITON_HN" == "yes" ]]; then
+		return
+	fi
+
+	OLD_DEFAULT="$DEFAULT_URL_PREFIX"
+	if [[ -f $PIADM_CONF ]]; then
+		. $PIADM_CONF
+	else
+		# We're creating $PIADM_CONF.
+		# NOTE: On an installation this will disappear as /var is
+		# on the ramdisk.
+		vecho "Creating $PIADM_CONF"
+		rm -rf $PIADM_CONF
+		mkdir -p /var/piadm
+
+		# The latest default behavior should be hardcoded in this path.
+		# Update as we add more things.  Each change bumps
+		# PIADM_CONFIG_VERSION by one.
+		PIADM_CONFIG_VERSION=1
+		printf "PIADM_CONFIG_VERSION=%s\n" $PIADM_CONFIG_VERSION > \
+			$PIADM_CONF
+		printf "DEFAULT_URL_PREFIX=%s\n" $DEFAULT_URL_PREFIX >> \
+			$PIADM_CONF
```

Comment:
Consider a heredoc instead of multiple redirects.

Reply:
Like:
```
cat <<<FILE
PIADM_CONFIG_VERSION=$PIADM_CONFIG_VERSION
...
FILE > $PIADM_CONF
```
?

Reply:
Just `<<` not `<<<`, but otherwise yeah.

```
cat << FILE
stuff
FILE > $PIADM_CONF
```

Reply:
Will fix.